sbi/utils/pyroutils.py

```diff
@@ -33,3 +40,151 @@ def prior():
             transforms[name] = biject_to(fn.support).inv
 
     return transforms
+
+
+class EstimatorDistribution(pyro.distributions.TorchDistribution):
+    """
+    Base class for a conditioned `sbi` estimator wrapped as a Pyro distribution.
+    """
+
+    def __init__(self, estimator: Any, condition: torch.Tensor):
+        self._estimator = estimator
+        condition_shape, event_shape = self.get_condition_and_event_shapes()
+        self._check_condition_shape(condition)
+        self._condition = condition
+        super().__init__(
+            batch_shape=condition.shape[: -len(condition_shape)],
+            event_shape=event_shape,
+        )
+
+    @abstractmethod
+    def get_condition_and_event_shapes(self) -> Tuple[torch.Size, torch.Size]:
+        pass
+
+    @property
+    def support(self) -> constraints.Constraint:
+        # Note: this will currently be wrong for estimators with discrete or mixed
+        # support (e.g. `MixedDensityEstimator`). Torch constraints currently don't
+        # support mixed support, and support is primarily used for choosing
+        # unconstraining transforms for priors.
```

Comment:
it would be good to have an issue for this. do you know a workaround for this problem, writing custom support for the `MixedDensityEstimator`? 
This would be nice to have becuase the mixed setting is a likely use-case, e.g., in decision-making models.

Reply:
I couldn't find one, so I created one here: pytorch/pytorch#149718